components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -1136,161 +975,79 @@ public void testImporterAsNoSudoChgrpChownWorkflow(boolean permChgrp, boolean pe
+         * and the link too. Also check that the canChown boolean on the Dataset must be in
+         * sync with the permChown.*/
+        Assert.assertEquals(getCurrentPermissions(sentDat).canChown(), permChown);
+        doChange(client, factory, Requests.chown().target(sentDat).toUser(normalUser.userId).build(), permChown);
+        /* boolean importYourGroupAndChgrpAndChownExpectSuccess
+         * captures permChown and permChgrp. Check the objects ownership and groups.*/
         if (importYourGroupAndChgrpAndChownExpectSuccess) {/* whole workflow2 succeeded */
-            /* Check the value of canChown on the dataset is true in this case.*/
-            Assert.assertTrue(getCurrentPermissions(sentDat).canChown());
-            doChange(client, factory, Requests.chown().target(sentDat).toUser(normalUser.userId).build(), true);
-            remoteFileGroupId = ((RLong) iQuery.projection(
-                    "SELECT details.group.id FROM OriginalFile o WHERE o.id = :id",
-                    new ParametersI().addId(remoteFile.getId())).get(0).get(0)).getValue();
-            imageGroupId = ((RLong) iQuery.projection(
-                    "SELECT details.group.id FROM Image i WHERE i.id = :id",
-                    new ParametersI().addId(image.getId())).get(0).get(0)).getValue();
-            datasetGroupId = ((RLong) iQuery.projection(
-                    "SELECT details.group.id FROM Dataset d WHERE d.id = :id",
-                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
-            long datasetImageLinkId = ((RLong) iQuery.projection(
-                    "SELECT id FROM DatasetImageLink WHERE parent.id = :id",
-                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
-            datasetImageLinkGroupId = ((RLong) iQuery.projection(
-                    "SELECT details.group.id FROM DatasetImageLink WHERE parent.id = :id",
-                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
             /* image, dataset and link are in the normalUser's group and belong to normalUser */
-            assertOwnedBy(remoteFile, normalUser);
-            Assert.assertEquals(remoteFileGroupId, normalUser.groupId);
+            assertOwnedBy(originalFile, normalUser);
+            assertInGroup(originalFile, normalUser.groupId);
             assertOwnedBy(image, normalUser);
-            Assert.assertEquals(imageGroupId, normalUser.groupId);
+            assertInGroup(image, normalUser.groupId);
             assertOwnedBy(sentDat, normalUser);
-            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertInGroup(sentDat, normalUser.groupId);
             assertOwnedBy((new DatasetImageLinkI(datasetImageLinkId, false)), normalUser);
-            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+            assertInGroup((new DatasetImageLinkI(datasetImageLinkId, false)), normalUser.groupId);
```

Comment:
maybe use `findByQuery` instead of `projection` to get the actual link then you can just pass that here, instead of fetching IDs then reconstructing objects?